pkg/platform/runtime2/setup/buildlog/buildlog.go

```diff
@@ -66,6 +65,7 @@ func New(artifactMap map[artifact.ArtifactID]artifact.ArtifactRecipe, conn Build
 		defer close(ch)
 		defer close(errCh)
 		defer messageHandler.BuildFinished()
+		defer conn.Close()
```

Comment:
This doesn't belong here imo. The connection was passed into this function, it should be closed by whatever created it.

Reply:
Another option is to actually create the connection here, in this function.

Reply:
Yes, you are right.  It should be closed outside.  I prefer providing the connection as an argument, so we can test the logic with a "connection" that does not have to talk to server.